cmd/ocsp-responder/main.go

```diff
@@ -217,17 +212,25 @@ as generated by Boulder's single-ocsp command.
 	go cmd.DebugServer(c.OCSPResponder.DebugAddr)
 	go cmd.ProfileCmd(scope)
 
-	hd := &httpdown.HTTP{
-		StopTimeout: stopTimeout,
-		KillTimeout: killTimeout,
-	}
-	hdSrv, err := hd.ListenAndServe(srv)
-	cmd.FailOnError(err, "Error starting HTTP server")
+	done := make(chan bool)
+	go cmd.CatchSignals(logger, func() {
+		ctx, cancel := context.WithTimeout(context.Background(),
+			c.OCSPResponder.ShutdownStopTimeout.Duration)
+		defer cancel()
+		_ = srv.Shutdown(ctx)
+		done <- true
+	})
 
-	go cmd.CatchSignals(logger, func() { _ = hdSrv.Stop() })
+	err = srv.ListenAndServe()
+	if err != nil && err != http.ErrServerClosed {
+		cmd.FailOnError(err, "Running HTTP server")
+	}
 
-	forever := make(chan struct{}, 1)
-	<-forever
+	// https://godoc.org/net/http#Server.Shutdown:
```

Comment:
Could you copy this comment to `wfe.go` too. Or it might make sense to just move this (L215-L233) into `cmd` so that we aren't duplicating it in two places.